File: Programming Basics/Programming Basics - C#/Exams/Programming Basics Exam - 17 July 2016/Exam - 17 July 2016/3. Match Tickets/Match Tickets.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _3.Match_Tickets
{
    class Program
    {
        static void Main(string[] args)
        {
            double budget = double.Parse(Console.ReadLine());
            string category = Console.ReadLine().ToLower();
            int people = int.Parse(Console.ReadLine());

            double moneyForTransport;
            double moneyLeft;
            double moneyNeededForTickets;


            if (people >= 1 && people <= 4)
            {
                moneyForTransport = budget * 0.75;
            }

            else if (people >= 5 && people <= 9)
            {
                moneyForTransport = budget * 0.60;
            }

            else if (people >= 10 && people <= 24)
            {
                moneyForTransport = budget * 0.50;
            }

            else if (people >= 25 && people <= 49)
            {
                moneyForTransport = budget * 0.40;
            }

            else
            {
                moneyForTransport = budget * 0.25;
            }

            moneyLeft = budget - moneyForTransport;

            if (category == "normal")
            {
                moneyNeededForTickets = 249.99 * people;
                double diff = Math.Abs(moneyLeft - moneyNeededForTickets);

                if (moneyLeft >= moneyNeededForTickets)
                {
                    Console.WriteLine("Yes! You have {0:f2} leva left.", diff);
                }
                else
                {
                    Console.WriteLine("Not enough money! You need {0:f2} leva.", diff);
                }
            }

            else if (category == "vip")
            {
                moneyNeededForTickets = 499.99 * people;
                double diff = Math.Abs(moneyLeft - moneyNeededForTickets);

                if (moneyLeft >= moneyNeededForTickets)
                {
                    Console.WriteLine("Yes! You have {0:f2} leva left.", diff);
                }
                else
                {
                    Console.WriteLine("Not enough money! You need {0:f2} leva.", diff);
                }
            }
        }
    }
}

File: Programming Basics/Programming Basics - JavaScript/Exams/Programming Basics Exam - 17 July 2016/3. Match Tickets/Match Tickets.js
function matchTickets([arg1, arg2, arg3]) {
    let budget = Number(arg1);
    let category = arg2;
    let people = Number(arg3);

    let budgetForTransport;
    let budgetForTickets;
    let budgetLeft;

    if (people <= 4) {
        budgetForTransport = budget * 0.75;
    }
    else if (people >= 5 && people < 10) {
        budgetForTransport = budget * 0.60;
    }
    else if (people >= 10 && people < 25) {
        budgetForTransport = budget * 0.50;
    }
    else if (people >= 25 && people < 50) {
        budgetForTransport = budget * 0.40;
    }
    else {
        budgetForTransport = budget * 0.25;
    }

    budgetLeft = budget - budgetForTransport;

    if (category == "VIP") {
        budgetForTickets = people * 499.99;
        let moneyLeft = Math.abs(budgetLeft - budgetForTickets);

        if (budgetLeft >= budgetForTickets) {
            console.log(`Yes! You have ${moneyLeft.toFixed(2)} leva left.`);
        }
        else {
            console.log(`Not enough money! You need ${moneyLeft.toFixed(2)} leva.`);
        }
    }

    else if (category == "Normal") {
        budgetForTickets = people * 249.99;
        let moneyLeft = Math.abs(budgetLeft - budgetForTickets);

        if (budgetLeft >= budgetForTickets) {
            console.log(`Yes! You have ${moneyLeft.toFixed(2)} leva left.`);
        }
        else {
            console.log(`Not enough money! You need ${moneyLeft.toFixed(2)} leva.`);
        }
    }
}

/* Uncomment to test:
matchTickets(["1000", "Normal", "1"]);
matchTickets(["30000", "VIP", "49"]);
    */
